csrc/include/natten/cuda/fna/kernel_forward.h

```diff
@@ -1004,6 +1015,16 @@ struct FusedNeighborhoodAttentionKernel {
         }
       }
     }
+    
+    if (p.maximums_ptr && thread_id() < kQueriesPerBlock){
+      constexpr float kLog2e = 1.4426950408889634074; // log_2(e) = M_LOG2E
+      auto query_idx =
+          map_index_to_coord((int32_t)thread_id(), problem_size_0_m);
+      auto query_offset = (query_idx * p.maximums_strideM).sum();
+      if (is_coord_within_upper_bound(query_idx, problem_size_0_m)) {
+        p.maximums_ptr[query_offset] = mi[thread_id()] / kLog2e; 
+      }
+    }
```

Comment:
Again a lot of code repetition here. Can we simplify and merge with the LSE block above?